src/atoms/ApmTableOfContents/ApmTableOfContents.js

```diff
@@ -23,29 +24,53 @@ const ApmTableOfContents = (props) => {
     );
   });
 
-  return (
-    <>
-      <ul className="table-of-contents">
-        {Headers.map((header) => {
-          return (
-            <li
-              className={`table-of-contents-level-${header.attrs.level}`}
-              key={uuid()}
-            >
-              <a
-                href={`#${header.content[0].text
-                  .toLowerCase()
-                  .replace(/[^a-z0-9+]+/gi, '')}.heading_${header.attrs.level}`}
+  if (toc.attrs.anchors) {
+    return (
+      <>
+        <ul className="table-of-contents">
+          {toc.attrs.anchors.map((anchor) => {
+            return (
+              <li
+                key={anchor.anchor}
+                className={`table-of-contents-level-${anchor.level}`}
               >
-                {header.content[0].text}
-              </a>
-            </li>
-          );
-        })}
-      </ul>
-      {theRest}
-    </>
-  );
+                <a href={`#${anchor.anchor}`}>{anchor.linkText}</a>
+              </li>
+            );
+          })}
+        </ul>
+        {theRest}
```

Comment:
Is there a particular reason to render the rest here vs just letting the outer Traverse method work the way it usually does?

Reply:
Good point  - seemed like a good idea at  the time.  I'll try it that way

Reply:
Yes now I see the reason I have to  sub in the HeadingWithTag component instead of using normal Heading. Should rename it to HeadingWithId   think I will do that

Reply:
Why not just always generate the heading ID?

Reply:
Well  fair point. In fact  I am doing that already but  it's missing  h1_  prefix  so I'll fix that and give this another look